client/kubernetes/src/main/java/org/jolokia/kubernetes/client/KubernetesJmxConnector.java

```diff
@@ -118,15 +119,13 @@ protected J4pClient expandAndProbeUrl(KubernetesClient client,
     throw new MalformedURLException("Unable to connect to proxypath " + proxyPath);
   }
 
-  private StringBuilder buildProxyPath(Pod pod, String protocol, String port, String path) {
-    final StringBuilder url = new StringBuilder(pod.getMetadata().getSelfLink());
+  public static StringBuilder buildProxyPath(Pod pod, String protocol, String port, String path) {
+    final ObjectMeta metadata = pod.getMetadata();
+    final StringBuilder url = new StringBuilder("/api/").append(pod.getApiVersion()).append("/namespaces/").append(metadata.getNamespace()).append("/pods/");
     if (protocol != null && !protocol.equals("http:")) {
-      //if we need to specify protocol, inject in url
-      final int index = url.lastIndexOf(pod.getMetadata().getName());
-      if (index > -1) {
-        url.insert(index, protocol);
-      }
+      url.append(protocol);
```

Comment:
Haven't seen the protocol in API URLs really before. Why do you need it here ?

Reply:
It allows you to specify https instead of http